main.tf

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  project      = "${data.google_project.project.project_id}"
+  external_ip  = "${var.external_ip}"
+  service_name = "${local.name}.endpoints.${local.project}.cloud.goog"
+  name         = "${var.name}"
+}
+
+data "google_project" "project" {
```

Comment:
Looks like all this `google_project` data source is used for is to fetch the Project ID (which you already have). Dropping it could save an API call.